meson.build

```diff
@@ -1,5 +1,5 @@
 project('qemu', ['c'], meson_version: '>=0.63.0',
-        default_options: ['warning_level=1', 'c_std=gnu11', 'cpp_std=gnu++11', 'b_colorout=auto',
+        default_options: ['warning_level=1', 'c_std=gnu11', 'cpp_std=gnu++17', 'b_colorout=auto',
```

Comment:
Is it ok to keep this? I couldn't get rid of this modification in the end

Reply:
Is it required because the SymCC runtime uses C++17? Fine for me to keep it :man_shrugging:

Reply:
It should not be necessary in theory.
Subprojects in meson should not rely on the options of the root project by default. I suspect the parsing of cmake by meson is doing something wrong, I will check

Reply:
After investigation, I don't think we can do much about this.
Apparently, a few things are being hardcoded in the default options of the meson file generated from the cmake project that cannot be avoided (https://github.com/mesonbuild/meson/blob/master/mesonbuild/cmake/interpreter.py).
Maybe I missed something. If someone finds a way, we can change. Otherwise, I'll keep things that way.